pkg/resource/rest.go

```diff
@@ -55,7 +55,11 @@ func GetAuthorizedRestConfig(apiServer, apiServerToken, caData string, tls rest.
 	return
 }
 
-func BuildRestConfig(apiServer, apiServerToken, caData string, tls rest.TLSClientConfig) (*rest.Config, error) {
+func BuildRestConfig(apiServer, apiServerToken, caData string, tls *rest.TLSClientConfig) (*rest.Config, error) {
+	if tls == nil {
+		return nil, fmt.Errorf("can't build a RestConfig from nil")
+	}
```

Comment:
We can allow nil - just use an empty struct as the caller would have done before.

```suggestion
	if tls == nil {
		tls = &rest.TLSClientConfig{}
	}
```